text/0000-lts.md

```diff
@@ -0,0 +1,164 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Two new channels: lts and lts-prior
+
+In addition to `stable`, `beta`, and `nightly`, we will add two additional
+release tracs:
+
+* `lts`: The most recent LTS release.
+* `lts-prior`: The second most recent LTS release.
+
+Every fourth stable/beta release, we will perform a release on both the `lts`
+and `lts-prior` channels. The **previous** stable release will be the basis for
+the new `lts` release, and the previous `lts` release will move to the
+`lts-prior` release. This is analogous to the way that the previous beta
+becomes the new stable.
+
+To keep it as easy as possible to remember which releases are lts release, we
+will begin the lts schedule so that the point version of the LTS release is
+always divisible by four. Depending on implementation schedule, that means that
+the first LTS release will probably be 1.28 or 1.32, release concurrently with
+either 1.29 or 1.33.
+
+## Level of support from the Rust project
+
+All three of the stable release channels - stable, lts, and lts-prior - will
+receive the same level of support. We will host documentation for all three
+channels and provide equivalent levels of patch release support - that is, if a
+bug justifying a patch exists in any of these channels, we will release a patch
+for that channel, just as we do today for stable. However, lts releases are
+less likely to receive patch releases in practice, since these bugs are most
+often found in the first six weeks after a stable release.
+
+As a result of this, a stable release which is a candidate for the lts channel
+(that is, its minor version is a multiple of four) will be supported across all
+three channels for the following nine release cycles, a period of 54 weeks or
+just over a year.
+
+## Recommendations for library authors
+
+Today, library authors fall into three general camps:
+
+1. Their library compiles on the nightly channel.
+2. Their library compiles on the stable channel.
+3. Their library compiles on any release more recent than an arbitrary old
+release, which is different for each library (for example, `failure` compiles
+on 1.18.0, whereas `serde` compiles on 1.13.0).
+
+Frequently, whichever level of support they offer, they guarantee it by testing
+their code on that channel or release using CI.
+
+The Rust project recommends that library authors interested in providing
+support for users on versions older than the latest stable, instead of focusing
+on supporting particular *versions*, focus on supporting particular *release
+channels*. That is, a library may guarantee that it will compile on any of the
+stable release channels, determining how old the version of Rust their users
+can be on with a guarantee of compatibility.
+
+We recommend that libraries consider dropping support for a release channel to
+be a breaking change, but that dropping support for any particular version of
+Rust to be a non-breaking change. 
```

Comment:
AFAIK, libraries have until now considered bumping the minimum stable release requirement to be a breaking change. This seems to weaken that guarantee considerably?

Reply:
@Centril [the question is contentious](https://github.com/rust-lang-nursery/api-guidelines/issues/123). This suggestion is somewhat of a middle ground.

Reply:
@est31 Thanks for the link -- I haven't read the thread in depth, but think I agree with @KodrAus.

Reply:
@Centril To summarize, I don't think there is actually any strong consensus on this point. Some libraries do as you say. Others adopt a different policy as a compromise where they are allowed to bump the minimum Rust version in minor version releases. Some libraries even bump the minimum Rust version in patch releases, although it is sometimes for good reason. This happened, in recent memory, with the `log` and `rand` crates, where they used the semver trick to make a patch release depend on a new minor release, which in turn required a newer rustc.

In my view, we're kind of all over the place right now. The best I've come to hope for is that library authors pin a Rust release in their CI so that it's at least easy to observe a possibly non-minimal lower bound.

I welcome a recommendation (such as the one in this RFC) that we can all unite around.

Reply:
@BurntSushi Alright; Thank you for the summary. The policy mess sounds like a bad place to be in, I agree. I agree that there should be a recommendation to unite around, but I don't think it should be the one in this RFC.